### PR TITLE
feat: associate contacts with posts

### PR DESCRIPTION
**Pull request by djaiss**
This PR lets you associate a contact with a post.

You can associate as many contacts in a post as you want.

This uses the `ContactSelector` component that is also used on the Loan widget.

### CODE REVIEW

**Comment by djaiss** [comment]
@asbiin I need your help. I don't understand what's happening inside the ContactSelector component.

Basically, inside the Edit view, the Contacts are passed as `ModelValue` to the ContactSelector component. Inside the component, the ModelValue instantiates an array called `localContacts` which is used to populate the list of contacts.

However, on page load, the contacts are not displayed, because the component thinks the localContacts variable is empty, even though ModelValue is not empty.

I have no idea what's going on and why this is happening.

**Comment by asbiin** [comment]
> However, on page load, the contacts are not displayed, because the component thinks the localContacts variable is empty, even though ModelValue is not empty.

@djaiss yes on page load == when vue run `mounted`, the value of the `modelValue` isn't set yet. You need to add a `watch` method on the `modelValue` to set the value of the local variable when `modelValue` changes.

File: app/Domains/Contact/ManageContact/Web/ViewHelpers/ContactShowViewHelper.php [hunk 1]
```diff
@@ -285,6 +286,10 @@ public static function modules(TemplatePage $page, Contact $contact, User $user)
                 $data = ModulePhotosViewHelper::data($contact);
             }
 
+            if ($module->type == Module::TYPE_POSTS) {
```
Comment (asbiin) [inline app/Domains/Contact/ManageContact/Web/ViewHelpers/ContactShowViewHelper.php:289]: I would have used a `switch` on the whole foreach, but it's more changes

File: app/Domains/Vault/ManageJournals/Services/AddContactToPost.php [hunk 2]
```diff
@@ -0,0 +1,97 @@
...
+
+        return $this->post;
+    }
+
+    private function validate(): void
+    {
+        $this->validateRules($this->data);
+
+        $journal = Journal::where('vault_id', $this->data['vault_id'])
+            ->findOrFail($this->data['journal_id']);
```
Comment (asbiin) [inline app/Domains/Vault/ManageJournals/Services/AddContactToPost.php]: Try
```suggestion
        $journal = $this->vault->journals()
            ->findOrFail($this->data['journal_id']);
```

File: app/Domains/Vault/ManageJournals/Services/AddContactToPost.php [hunk 3]
```diff
@@ -0,0 +1,97 @@
...
+
+        $this->post = Post::where('journal_id', $journal->id)
+            ->findOrFail($this->data['post_id']);
```
Comment (asbiin) [inline app/Domains/Vault/ManageJournals/Services/AddContactToPost.php]:
```suggestion
        $this->post = $journal->posts()
            ->findOrFail($this->data['post_id']);
```

File: app/Domains/Vault/ManageJournals/Services/RemoveContactFromPost.php [hunk 4]
```diff
@@ -0,0 +1,99 @@
...
+
+    private function validate(): void
+    {
+        $this->validateRules($this->data);
+
+        Journal::where('vault_id', $this->data['vault_id'])
+            ->findOrFail($this->data['journal_id']);
+
+        $this->post = Post::where('journal_id', $this->data['journal_id'])
+            ->findOrFail($this->data['post_id']);
```
Comment (asbiin) [inline app/Domains/Vault/ManageJournals/Services/RemoveContactFromPost.php]:
```suggestion
        $journal = $this->vault->journals()
            ->findOrFail($this->data['journal_id']);

        $this->post = $journal->posts()
            ->findOrFail($this->data['post_id']);
```

**Review by asbiin — APPROVED** [review]
I'm proposing some small changes, but otherwise it's great!

File: app/Domains/Contact/ManageContact/Web/ViewHelpers/ContactShowViewHelper.php (hunk 1 above)
Comment (djaiss) [reply app/Domains/Contact/ManageContact/Web/ViewHelpers/ContactShowViewHelper.php:289]: You are right. I won't do this in this PR but will do it. It'll be faster.

**Pull request merged**